packages/jitm/src/JITM.php

```diff
@@ -149,6 +153,152 @@ static function jitm_jetpack_woo_services_activate( $CTA ) {
 		);
 	}
 
+	/**
+	 * This is an entire admin notice dedicated to messaging and handling of the case where a user is trying to delete
+	 * the connection owner.
+	 */
+	function delete_user_update_connection_owner_notice() {
+		global $current_screen;
+
+		if ( ! isset( $current_screen->base ) || 'users' !== $current_screen->base ) {
+			return;
+		}
+
+		if ( ! isset( $_REQUEST['action'] ) || 'delete' !== $_REQUEST['action'] ) {
+			return;
+		}
+
+		// Get connection owner or bail
+		$connection_manager = new Manager();
+		$connection_owner   = $connection_manager->get_connection_owner();
+		if ( ! $connection_owner ) {
+			return;
+		}
+
+		// Bail if we're not trying to delete connection owner.
+		$user_ids_to_delete = array();
+		if ( isset( $_REQUEST['users'] ) ) {
+			$user_ids_to_delete = $_REQUEST['users'];
+		} elseif ( isset( $_REQUEST['user'] ) ) {
+			$user_ids_to_delete[] = $_REQUEST['user'];
+		}
+		$deleting_connection_owner = in_array( $connection_owner->ID, (array) $user_ids_to_delete );
+		if ( ! $deleting_connection_owner ) {
+			return;
+		}
+
+		// Bail if they're trying to delete themselves to avoid confusion.
+		if ( $connection_owner->ID == get_current_user_id() ) {
+			return;
+		}
+
+		$connection_manager = new Manager();
+		$connected_users    = $connection_manager->get_connected_users( 'jetpack_connect' );
```

Comment:
Should it be `$connnected_admins` here, since we are only looking for users that can connect Jetpack to WordPress.com for the site? I was a bit confused at first.